File: R/rdirichlet.R
#' Generate a random draw from a Dirichlet distribution
#'
#' @description
#' Draws a single random vector from a Dirichlet distribution using the gamma trick.
#'
#' @param alpha Numeric vector; concentration parameters of the Dirichlet (length K).
#'
#' @return Numeric vector of length K that sums to 1, a single Dirichlet(α) draw.
#'
#' @examples
#' rdirichlet(c(1, 1, 1))  # uniform over 3-simplex
#'
#' @export
rdirichlet <- function(alpha) {
  x <- rgamma(length(alpha), shape = alpha, rate = 1)
  x / sum(x)
}

File: nicknamer/rdirichlet.py
"""Draw a single sample from a Dirichlet distribution."""

import numpy as np
from typing import Union


def rdirichlet(alpha: Union[np.ndarray, list]) -> np.ndarray:
    """Draw a single sample from Dirichlet(alpha) using the gamma trick.

    Parameters
    ----------
    alpha : array-like of shape (K,)
        Concentration parameters (all strictly positive).

    Returns
    -------
    numpy.ndarray of shape (K,)
        A probability vector (sums to 1).
    """
    alpha = np.asarray(alpha, dtype=float)
    x = np.random.gamma(shape=alpha, scale=1.0)
    return x / x.sum()
